File: py/941.py
import unittest


class Solution:
    def validMountainArray(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """
        climb = True

        for i in range(1, len(A)):
            if A[i-1] == A[i]:
                return False

            if climb:
                if A[i-1] > A[i]:
                    if i == 1:
                        return False
                    else:
                        climb = False
            else:
                if A[i-1] <= A[i]:
                    return False

        return not climb


class SolutionTest(unittest.TestCase):
    def test_validMountainArray(self):
        s = Solution()
        self.assertTrue(s.validMountainArray([0, 3, 2, 1]))
        self.assertFalse(s.validMountainArray([2, 1]))
        self.assertFalse(s.validMountainArray([3, 5, 5]))

if __name__ == "__main__":
    unittest.main()

File: js/941.js
/**
 * @param {number[]} A
 * @return {boolean}
 */
var validMountainArray = function (A) {

  let climb = true
  for (let i = 1; i < A.length; i++) {
    if (climb) {
      if (A[i - 1] === A[i]) {
        return false
      } else if (A[i - 1] > A[i]) {
        if (i === 1) {
          return false
        }
        climb = false
      }
    } else {
      if (A[i - 1] <= A[i]) {
        return false
      }
    }
  }
  return !climb
};

var assert = require('assert');

assert.ok(validMountainArray([0, 3, 2, 1]))
assert.ok(!validMountainArray([2, 1]))
assert.ok(!validMountainArray([3, 5, 5]))